org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/variables/IVariableFormatter.java

```diff
@@ -38,6 +38,15 @@
     void registerValueFormatter(IValueFormatter formatter, int priority);
 
     /**
+     * Get the matched value formatter for the specified type.
+     * @param type the type of the valuer
+     * @param options the options about the formatters
+     * @return the matched value formatter, if no formatter matches,
+     * <code>IllegalArgumentException</code> will be thrown
+     */
+    IValueFormatter getValueFormatter(Type type, Map<String, Object> options);
```

Comment:
added public Value stringToValue(String stringValue, Type type, Map<String, Object> options)  to VariableFormatter